### PR TITLE
Added architecture detection

### PR DESCRIPTION
**Pull request by tonk**
As a first attempt to code in Rust :-)
I added remote host architecture detection

### CODE REVIEW

File: src/modules/control/facts.rs [hunk 1]
```diff
@@ -137,5 +138,16 @@ impl FactsAction {
...
+
+    fn do_linux_os_arch(&self, handle: &Arc<TaskHandle>, request: &Arc<TaskRequest>, mapping: &Arc<RwLock<serde_yaml::Mapping>>) -> Result<(), Arc<TaskResponse>> {
+        // run uname -m to get the architecture of the system
+        let cmd = String::from("uname -m");
+        let result = handle.remote.run(request, &cmd, CheckRc::Checked)?;
+        let (_rc, out) = cmd_info(&result);
+        for line in out.lines() {
+            self.insert_string(mapping, &String::from("jet_os_arch"), &String::from(line));
+        }
+        return Ok(());
```
Comment (pcarinhas) [inline src/modules/control/facts.rs:150]: This should just be:

`Ok(())`

**Review by pcarinhas — CHANGES_REQUESTED** [review]

File: src/modules/control/facts.rs [hunk 2]
```diff
@@ -137,5 +138,16 @@ impl FactsAction {
         }
         return Ok(());
     }
+
+    fn do_linux_os_arch(&self, handle: &Arc<TaskHandle>, request: &Arc<TaskRequest>, mapping: &Arc<RwLock<serde_yaml::Mapping>>) -> Result<(), Arc<TaskResponse>> {
+        // run uname -m to get the architecture of the system
+        let cmd = String::from("uname -m");
+        let result = handle.remote.run(request, &cmd, CheckRc::Checked)?;
+        let (_rc, out) = cmd_info(&result);
+        for line in out.lines() {
```
Comment (pcarinhas) [inline src/modules/control/facts.rs:147]: Why does uname have mulitple lines?

File: src/modules/control/facts.rs (hunk 1 above)
Comment (tonk) [reply src/modules/control/facts.rs:150]: OK. I just "stole" that from some lines up :-)

File: src/modules/control/facts.rs (hunk 2 above)
Comment (tonk) [reply src/modules/control/facts.rs:147]: Very fair question.

It doesn't, but as I already told @mpdehaan my Rust skills are not up to scratch yet. I couldn't find a way to get a single line out of `out.lines()` any other way. So I copied the code from above,

File: src/modules/control/facts.rs (hunk 1 above)
Comment (mpdehaan) [reply src/modules/control/facts.rs:150]: @pcarinhas please leave pull request review to me, thank you

Comment (mpdehaan) [reply src/modules/control/facts.rs:150]: @tonk - that return is fine, I think it increases readability and I'm not worried about it.

**Review by mpdehaan — CHANGES_REQUESTED** [review]

**Comment by tonk** [comment]
I have added the `line.trim()` .
Do you want a rebase on this?

File: src/modules/control/facts.rs [hunk 3]
```diff
@@ -145,7 +145,8 @@ impl FactsAction {
         let result = handle.remote.run(request, &cmd, CheckRc::Checked)?;
         let (_rc, out) = cmd_info(&result);
         for line in out.lines() {
```
Comment (mpdehaan) [inline src/modules/control/facts.rs:147]: remove for loop

**Review by mpdehaan — CHANGES_REQUESTED** [review]

**Comment by mpdehaan** [comment]
I think it's been too long since I've used git so I don't really see a point in the rebase, Squashing it into one commit would be nice

Looks like you still have the for loop in there though, the suggestion was to get rid of the for loop?  If that's gone, I apologize and it's the result

**Comment by tonk** [comment]
Removed the `for` loop

**Comment by mpdehaan** [comment]
merged in, didn't need the trim after all as the code removes the last newline in the connection, also arch is now returned on OS X as well.

Thanks!!!!

**Pull request merged**